### PR TITLE
Add isoperation and isoperationtype

### PR DESCRIPTION
**Pull request by mforets**
See #991.

### CODE REVIEW

File: src/Interfaces/LazySet.jl [hunk 1]
```diff
@@ -588,3 +589,36 @@ function plot_recipe(X::LazySet{N}, ε::N=N(PLOT_PRECISION)) where {N<:Real}
     end
     return plot_recipe(Y, ε)
 end
+
+"""
+    isoperation(::Type{<:LazySet})
+
+Check whether the given lazy set is an operation or not.
```
Comment (schillic) [inline src/Interfaces/LazySet.jl]:
```suggestion
Check whether the given `LazySet` is an operation or not.
```

File: src/LazyOperations/ExponentialMap.jl [hunk 2]
```diff
@@ -60,6 +60,8 @@ struct SparseMatrixExp{N, MN<:AbstractSparseMatrix{N}} <: AbstractMatrix{N}
     M::MN
 end
 
+isoperation(::Type{SparseMatrixExp}) = true
```
Comment (schillic) [inline src/LazyOperations/ExponentialMap.jl]: Was this on purpose? `SparseMatrixExp` is not actually a set (same for `ProjectionSparseMatrixExp`).

**Comment by mforets** [comment]
> UnionSet/UnionSetArray are missing.

i only considered subtype of LazySet, but we can add these too

**Comment by schillic** [comment]
> i only considered subtype of LazySet, but we can add these too

No, you also modified `Complement`.

File: src/LazyOperations/ExponentialMap.jl (hunk 2 above)
Comment (mforets) [reply src/LazyOperations/ExponentialMap.jl]: true, this should be removed (same for the projection)

**Comment by mforets** [comment]
ok, so ill add them as well

**Comment by mforets** [comment]
> I would have expected that you create the method for the instances. I would definitely define the following methods:

or add `isoperation` and `isoperationtype`

File: src/Sets/Ball1.jl [hunk 3]
```diff
@@ -53,6 +53,8 @@ struct Ball1{N<:Real} <: AbstractCentrallySymmetricPolytope{N}
     end
 end
 
+isoperationtype(::Type{Ball1}) = false
```
Comment (schillic) [inline src/Sets/Ball1.jl]: Should this not be
```suggestion
isoperationtype(::Type{<:Ball1}) = false
```
(holds for all concrete set types)
```julia
julia> typeof(rand(BallInf))
BallInf{Float64}

julia> typeof(rand(BallInf)) isa Type{BallInf}
false

julia> typeof(rand(BallInf)) isa Type{<:BallInf}
true
```

Comment (mforets) [reply src/Sets/Ball1.jl]: yes, this is an omission from the concrete set types (i only added it in the operations)

**Comment by mforets** [comment]
Thanks for the review, im merging this.

**Pull request merged**